api/v1alpha4/types.go

```diff
@@ -116,6 +120,28 @@ type SubnetFilter struct {
 	NotTagsAny      string `json:"notTagsAny,omitempty"`
 }
 
+type PortOpts struct {
+	NetworkID           string              `json:"networkId" required:"true"`
+	NameSuffix          string              `json:"nameSuffix" required:"true"`
+	Description         string              `json:"description,omitempty"`
+	AdminStateUp        *bool               `json:"adminStateUp,omitempty"`
+	MACAddress          string              `json:"macAddress,omitempty"`
+	FixedIPs            []ports.IP          `json:"fixedIPs,omitempty"`
+	DeviceID            string              `json:"deviceID,omitempty"`
+	DeviceOwner         string              `json:"deviceOwner,omitempty"`
+	TenantID            string              `json:"tenantID,omitempty"`
+	ProjectID           string              `json:"projectID,omitempty"`
+	SecurityGroups      *[]string           `json:"securityGroups,omitempty"`
+	AllowedAddressPairs []ports.AddressPair `json:"allowedAddressPairs,omitempty"`
+
+	// The ID of the host where the port is allocated
+	HostID string `json:"binding:host_id,omitempty"`
```

Comment:
```suggestion
	HostID string `json:"binding:hostID,omitempty"`
```
I would prefer to stick to camel case